Source/NSubstitute/Routing/RouteFactory.cs

```diff
@@ -67,8 +67,8 @@ public IRoute RecordCallSpecification(ISubstituteState state)
         public IRoute RecordReplay(ISubstituteState state)
         {
             return new Route(RouteType.RecordReplay, new ICallHandler[] {
-                new ClearUnusedCallSpecHandler(state)
```

Comment:
Well, that makes the performance penalty because `ThreadLocal` works slower than [ThreadStatic] attribute, but I've changed that.